src/common/AndroidUtils.ts

```diff
@@ -644,35 +683,39 @@ export class AndroidSDKUtils {
         return AndroidSDKUtils.emulatorCommand;
     }
 
-    private static getAndroidToolsBin(): string {
-        if (!AndroidSDKUtils.androidToolsBin) {
-            const sdkRoot = AndroidSDKUtils.getAndroidSdkRoot();
-            AndroidSDKUtils.androidToolsBin = path.join(
-                (sdkRoot && sdkRoot.rootLocation) || '',
-                'tools',
-                'bin'
-            );
-
-            if (
-                !fs.existsSync(AndroidSDKUtils.androidToolsBin) &&
-                fs.existsSync(AndroidSDKUtils.getAndroidCmdLineToolsBin())
-            ) {
-                AndroidSDKUtils.androidToolsBin = AndroidSDKUtils.getAndroidCmdLineToolsBin();
-            }
-        }
-
-        return AndroidSDKUtils.androidToolsBin;
-    }
-
     private static getAndroidCmdLineToolsBin(): string {
         if (!AndroidSDKUtils.androidCmdLineToolsBin) {
             const sdkRoot = AndroidSDKUtils.getAndroidSdkRoot();
             AndroidSDKUtils.androidCmdLineToolsBin = path.join(
                 (sdkRoot && sdkRoot.rootLocation) || '',
-                'cmdline-tools',
-                'latest',
-                'bin'
+                'cmdline-tools'
             );
+
+            // It is possible to install various versions of the command line tools side-by-side
+            // In this case the directory structure would be based on tool versions:
+            //
+            //    cmdline-tools/1.0/bin
+            //    cmdline-tools/2.1/bin
+            //    cmdline-tools/3.0/bin
+            //    cmdline-tools/4.0-beta01/bin
+            //    cmdline-tools/latest/bin
+            //
+            // Below, we get the list of all directories, then sort them descendingly and grab the first one.
```

Comment:
I'm not sure we should do this, as IMO it gets tricky to decide what the user would actually prefer, and/or what we support. I'd advocate for continuing to use "latest", and telling the user to install the latest (that's what AS defaults to installing) if they haven't done so.

Reply:
So in AS if you expand "Android SDK Command-line Tools" you'd see this:

![Screen Shot 2021-01-19 at 9 31 58 AM](https://user-images.githubusercontent.com/65030660/105086369-c6d8a300-5a4d-11eb-8e50-a6d3ca417ce8.png)

As you can see there are 2 options for the latest version. If the user chooses "Android SDK Command-line Tools (latest)" then version 4.0.0 rc1 would be installed in a folder titled `latest` but if the user chooses the second option then it would be installed in a folder titled `4.0.0 rc1`.

If we told users that we recommend using the latest version, and if they have had chosen the second one from the above list then technically they do have the latest version but our tools would still not work b/c it would expect the folder name to be `latest`, which could cause confusion for the users.

For this reason I decided to add this logic here, to pick the latest version available. I've tested our command and they work on 1.0, 2.1, 3.0, and 4.0.0 rc1